src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -1424,6 +1424,8 @@ namespace QgsWms
     //then set destinationCrs
     mapSettings.setDestinationCrs( outputCRS );
 
+    mapSettings.setTransformContext( QgsProject::instance()->transformContext() );
```

Comment:
Elsewhere in this method I see `mProject` used, not QgsProject::instance.

Reply:
```suggestion
    mapSettings.setTransformContext( mProject->transformContext() );
```
yes: that works the same. A project pointer is also stored in wms context object, we have plenty of choices.